src/markets/search_and_matching.jl

```diff
@@ -550,34 +525,55 @@ function perform_retail_market!(
                     C_d_hg_[h] -= S_fg_[f] * P_f[f]
                     S_fg[f] -= S_fg_[f]
                     S_fg_[f] = 0
-                    deleteat!(F_g, e)
-                    isempty(F_g) && break
-                    pr_price_f = pos!(exp.(-2 .* @view(P_f[F_g])) ./ sum(exp.(-2 .* @view(P_f[F_g]))))
-                    pr_size_f = @view(S_f[F_g]) ./ sum(@view(S_f[F_g]))
-                    pr_cum_f_ = (pr_price_f + pr_size_f) ./ sum(pr_price_f + pr_size_f)
+                    delete!(F_g_sampler, e)
+                    isempty(F_g_sampler) && break
                 end
             end
-            H_g = findall(C_d_hg_ .> 0)
+            H_g = H_g[@view(C_d_hg_[H_g]) .> 0]
         end
     end
 
+    F_g = F_g[allinds(F_g_sampler)]
+
     a = @view(C_real_hg[1:H])
-    b = C_d_h .* b_HH_g[g] .- pos!(@view(C_d_hg[1:H]) .- b_CFH_g[g] .* I_d_h)
-    c = C_d_h .* b_HH_g[g] .+ b_CFH_g[g] .* I_d_h .- @view(C_d_hg[1:H])
-    d = pos!(b_CFH_g[g] .* I_d_h .- @view(C_d_hg[1:H]))
+    b = @~ C_d_h .* b_HH_g[g] .- pos.(@view(C_d_hg[1:H]) .- b_CFH_g[g] .* I_d_h)
```

Comment:
`@~`  comes from `LazyArrays.jl` and allow us to "wait" until that operations are actually required to run, so that we can fuse the operations, so that we don't need more passes over data